hcl2template/types.hcl_provisioner.go

```diff
@@ -40,6 +40,8 @@ func (p *HCL2Provisioner) HCL2Prepare(buildVars map[string]interface{}) error {
 			switch v := v.(type) {
 			case string:
 				buildValues[k] = cty.StringVal(v)
+			case uint8:
```

Comment:
I was thinking that maybe it should be better to replace this entire switch with https://github.com/hashicorp/packer-plugin-sdk/blob/main/hcl2helper/values.go#L22. It addresses all the config to cty.Value possible transformation.

Reply:
Eh, thats a good idea but `HCL2ValueFromConfigValue` panics and doesn't handle the uint8/int64/uint64 cases. 
Also, to not have to do two PRs each time we have this issue, I'd implement that one around here: https://github.com/hashicorp/packer/blob/291121dd5524d3b892a6d2a288531ffc8532dd86/hcl2template/shim/values.go#L18